src/spikeinterface/sorters/external/mountainsort5.py

```diff
@@ -186,21 +185,27 @@ def _run_from_folder(cls, sorter_output_folder, params, verbose):
             block_sorting_parameters=scheme2_sorting_parameters, block_duration_sec=p["scheme3_block_duration_sec"]
         )
 
-        with TemporaryDirectory(dir=p["temporary_base_dir"]) as tmpdir:
-            # cache the recording to a temporary directory for efficient reading (so we don't have to re-filter)
-            recording_cached = create_cached_recording(
-                recording=recording, folder=tmpdir, n_jobs=p["n_jobs_for_preprocessing"]
-            )
-
-            scheme = p["scheme"]
-            if scheme == "1":
-                sorting = ms5.sorting_scheme1(recording=recording_cached, sorting_parameters=scheme1_sorting_parameters)
-            elif p["scheme"] == "2":
-                sorting = ms5.sorting_scheme2(recording=recording_cached, sorting_parameters=scheme2_sorting_parameters)
-            elif p["scheme"] == "3":
-                sorting = ms5.sorting_scheme3(recording=recording_cached, sorting_parameters=scheme3_sorting_parameters)
-            else:
-                raise ValueError(f"Invalid scheme: {scheme} given. scheme must be one of '1', '2' or '3'")
+        if not recording.is_binary_compatible():
+            recording_cached = recording.save(folder=sorter_output_folder / "recording", **get_job_kwargs(p, verbose))
+        else:
+            recording_cached = recording
+
+        scheme = p["scheme"]
+        if scheme == "1":
+            sorting = ms5.sorting_scheme1(recording=recording_cached, sorting_parameters=scheme1_sorting_parameters)
+        elif p["scheme"] == "2":
+            sorting = ms5.sorting_scheme2(recording=recording_cached, sorting_parameters=scheme2_sorting_parameters)
+        elif p["scheme"] == "3":
+            sorting = ms5.sorting_scheme3(recording=recording_cached, sorting_parameters=scheme3_sorting_parameters)
```

Comment:
I know this was copy paste from the old way, but meant to ask is there a reason we want to do it this way? couldn't we just delete the scheme = p['scheme'] and use p['scheme'] for everything to be consistent?

Reply:
will do